src/compiler/scanner.rs

```diff
@@ -1,5 +1,44 @@
+#![allow(non_upper_case_globals)]
+
+use std::array::IntoIter;
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
 use crate::{CharacterCodes, SyntaxKind};
 
+lazy_static! {
+    static ref text_to_keyword_obj: HashMap<&'static str, SyntaxKind> =
+        HashMap::from_iter(IntoIter::new([
+            ("false", SyntaxKind::FalseKeyword),
+            ("true", SyntaxKind::TrueKeyword),
+        ]));
+}
+
+lazy_static! {
+    static ref text_to_keyword: HashMap<&'static str, SyntaxKind> = text_to_keyword_obj.clone();
```

Comment:
In the Typescript version `textToKeywordObj` is a JS object while `textToKeyword` is an ES6 `Map`, so here they're just both `HashMap`'s